src/reconciler/exec/io.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: MIT
+#![allow(unused_imports)]
+use crate::kubernetes_api_objects::{api_method::*, common::*};
+use crate::reconciler::spec::io::*;
+use vstd::{prelude::*, view::*};
+
+verus! {
+
+// Third-party libraries can also receive requests from reconciler.
+// T: The input type of the third-party library of the reconciler which should also be defined by the developer.
+// Typically, T can be an enum type, which lists all the possible supporting handlings the developer need support from the
+// third-party library on.
+// Then in the process method of the library, th developers can do pattern matching to generate desired output which will
+// then be sent to the reconciler in the next-round reconcile loop.
+// In reconcile_core, if the reconciler want kubernetes to process the request, it should return a Request::KRequest;
+// if it want the third-party library to deal with the request, it should return a Request::ExternalRequest.
+pub enum Request<T: View> {
+    KRequest(KubeAPIRequest),
+    ExternalRequest(T),
+}
+
+// The response type should be consistent with the Request type.
+// T: the output type of the third-party library of the reconciler which should be defined by the developer.
+// A safe and common way is to have an enum type which has the corresponding types of the input type in the Request.
+// Anway, the process method in the ExternalLibrary, the input type in Request, output type in Response and the handling
```

Comment:
Anyway?

Reply:
What should it be? I mean, although I introduce concretely what `T` is for, it is up to the developers to define it together with the Lib, reconcile_core process.

Reply:
oh I was just saying there is a typo